### PR TITLE
Zephyr Module

### PR DESCRIPTION
**Pull request by XenuIsWatching**
This creates the cmake with in the cmsis-nn for building as a zephyr module

### CODE REVIEW

**Comment by XenuIsWatching** [comment]
rebasing on top of cmsis-nn tag `v4.1.0`

File: zephyr/module.yml
Comment (JordanYates) [inline zephyr/module.yml:1]: I think it makes more sense to leave the build system in the repo intact (but ignored) and instead do our Zephyr thing in the main repo:
```
name: cmsis-nn
build:
  cmake-ext: True
  kconfig-ext: True
```
This should make updating the repo much simpler in the future by reducing the diff

**Review by JordanYates — APPROVED** [review]

**Comment by XenuIsWatching** [comment]
darn.... when merging this in.. it made a merge commit >:(

**Comment by JordanYates** [comment]
Its not ideal, but given my preference for rebasing our changes on top of upstream when updating it will disappear with the next update.

I'm more interested in why it happened in the first place (also why you were able to merge it as the author, which I thought was blocked by Github for Zephyr)

**Pull request merged**